kusto/query_builder.go

```diff
@@ -64,147 +65,146 @@ type ParamType struct {
 	name string
 }
 
-func (p ParamType) validate() error {
-	if !p.Type.Valid() {
-		return fmt.Errorf("the .Type was not a valid value, must be one of the values in this package starting with CT<type name>, was %s", p.Type)
-	}
-	if p.Default == nil {
-		return nil
+// isValid validates whether the value is of known kusto Types.
+func isValid(kustoType types.Column, val interface{}) bool {
+	if !kustoType.Valid() {
+		return false
 	}
 
-	switch p.Type {
+	switch kustoType {
 	case types.Bool:
-		if _, ok := p.Default.(bool); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(bool)
+		return ok
 	case types.DateTime:
-		if _, ok := p.Default.(time.Time); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(time.Time)
+		return ok
 	case types.Dynamic:
-		return fmt.Errorf("the .Type was %s, but Dynamic types cannot have default values", p.Type)
+		return false
 	case types.GUID:
-		if _, ok := p.Default.(uuid.UUID); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(uuid.UUID)
+		return ok
 	case types.Int:
-		if _, ok := p.Default.(int32); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
+		switch val.(type) {
+		case int32:
+			return true
+		case int:
+			return true
+		default:
+			return false
 		}
-		return nil
 	case types.Long:
-		if _, ok := p.Default.(int64); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(int64)
```

Comment:
Aren't int & int32 also valid to convert to long?

Reply:
based on what i've seen (and in the old tests as well) - it semms like Long is int64. am i missing something?